LIPS/lip-3.md

```diff
@@ -0,0 +1,152 @@
+
+#### `IncreaseNodeOperatorStakingLimit` script factory contract
+
+`IncreaseNodeOperatorStakingLimit` contract creates EVM scripts to increase node operator's staking limit. Requires permissions to call `setStakingLimit()` method on `NodeOperatorsRegistry` contract.
+To successfully complete the EVM script generation, the following requirements must be met:
+- reward address of the node operator matches the address of the caller;
+- Node Operator is not disabled;
+- the new staking limit is greater than the current staking limit;
+- total amount of signing keys submitted is less than or equal to the new staking limit.
+
+#### `TopUpLegoProgram` script factory contract
+
+`TopUpLegoProgram` contract creates EVM Script to top up the address of the LEGO program. Allows transferring various ERC20 tokens and ETH.
+
+#### `AddRewardProgram` script factory contract
+
+`AddRewardProgram` contract creates EVM Script to add new reward program address to `RewardProgramsRegistry`. 
+`RewardProgramsRegistry` contract is an additional contract storing a whitelist of allowed reward program addresses.
+To successfully complete the EVM script generation, the following requirements must be met:
+- `RewardProgramsRegistry` doesn’t already contain reward program address.
+
+#### `RemoveRewardProgram` script factory contract
+
+`RemoveRewardProgram` contract creates EVM Script to remove an existing reward program address from `RewardProgramsRegistry`. 
+To successfully complete the EVM script generation, the following requirements must be met:
+- `RewardProgramsRegistry` already contains the reward program address.
+
+#### `TopUpRewarProgram` script factory contract
+
+`TopUpRewarProgram` contract creates EVM Script to top up balances of reward programs. Addresses to be topped up are restricted to the list stored in the `RewardProgramsRegistry` contract.
+
+## `EVMScriptExecutor` contract
+
+The EVM scripts generated by the script factory contracts can be executed by a separate `EVMScriptExecutor` contract. It implements only `executeEVMScript()` method and accepts scripts built following the Aragon's EVM script structure. Current implementation uses [CallsScript.sol](https://github.com/aragon/aragonOS/blob/next/contracts/evmscript/executors/CallsScript.sol) contract of the Aragon framework. 
+`EVMScriptExecutor` permissions list is the same as the one of Aragon Voting App. While sharing the EVM script structure, both `EasyTrack` contract and Aragon Voting App are able to execute scripts using `EVMScriptExecutor` contract.
+___
+# Sanity and security
+
+The proposed Easy Track Motions design involves timelock as a native security feature. Any suspicious motion can be rejected or canceled within a relatively long motion duration period. To prevent malicious actions, we propose putting the following additional limitations in place (numeric values fixed based on community feedback from v1):
```

Comment:
Do we have a link on the feedback?

Reply:
Most of this discussion happened on research forum [here](https://research.lido.fi/t/lip-3-easy-track-motions-for-dao-routine-operations/680).
We have a discussion link in the LIP header already, but I am adding it here as well.